### ISSUE TITLE
[Web] Images are rotated in chrome browser with Canvaskit

### ISSUE THREAD

**Issue by OleFriedrich** [flutter/flutter#106342]
When building our application in flutter 3.0.x for web, some images are rotated by either 90 degrees or 180 degrees when using the canvasKit renderer. This only seems to be an issue in chromium based browsers (tested in chrome and edge), we also tested safari and firefox and everything was displayed properly. The issue does not occur when building with flutter 2.x versions. Also, it does not occur on mobile. Also, not every image is rotated. It seems to have something to do with the "orientation" EXIF data being set (if it is set, the images are rotated, if not, the images are displayed correctly).

## Steps to Reproduce
1. create an empty flutter 3.0.x project and put two simple AssetImage widgets in your widget tree
2. include the two attached sample images as assets in your project and display them 
3. build for chrome with canvasKit

**Expected results:**
Both images should be displayed correctly (in fact, if you view the same localhost page in a non-chromium browser, they are).

**Actual results:**
One image is rotated (with exif).

Flutter doctor:
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.0.2, on macOS 12.1 21C52 darwin-arm, locale de-DE)
    • Flutter version 3.0.2 at /Users/ole/Flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (13 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/ole/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.68.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.1 21C52 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 102.0.5005.115

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

Sample images
with exif          |  without exif
:-------------------------:|:-------------------------:
![with exif](https://user-images.githubusercontent.com/44000354/174823127-962e18da-ed25-4e1d-91f2-a810a18ad1a2.jpg) |  ![without exif](https://user-images.githubusercontent.com/44000354/174823189-adca7922-080e-455f-8b03-9bdea853bfc3.jpg)

**Comment by Yahllil** [comment]
Any update on this one? Flutter Web rotates the image for me as well

**Comment by SeanOThomas** [comment]
I'm seeing this issue on latest stable 3.0.2 for Chrome as well.

**Comment by maheshmnj** [comment]
Hi @OleFriedrich, Thanks for filing the issue.  I am able to reproduce the issue on both stable and the master channel. The image with exif label is rotated 90° anti-clockwise in Canvaskit, While this renders fine in HTML and on other platforms.

### Chrome
Canvaskit             |  HTML
:-------------------------:|:-------------------------:
<img width="612" alt="image" src="https://user-images.githubusercontent.com/31410839/174970499-52dcf625-464e-43e5-8ac3-6f46b637229c.png"> | <img width="612" alt="image" src="https://user-images.githubusercontent.com/31410839/174970880-f6d8e650-0b60-4728-82b8-722cee02d1f2.png">

### Firefox (Canvaskit/HTML)

<img src="https://user-images.githubusercontent.com/31410839/174972431-180e2015-ea5c-45d8-8121-9d97f82b083a.png" width ="300"> 

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;
  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: SingleChildScrollView(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Image(
              image: AssetImage('assets/dash.png'),
            ),
            Image(
              image: AssetImage('assets/withoutexif.jpeg'),
            ),
            Image(
              image: AssetImage('assets/withexif.jpeg'),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}

```
</details> 

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.2, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.2 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cd41fdd495 (5 days ago), 2022-06-08 09:52:13 -0700
    • Engine revision f15f824b57
    • Dart version 2.17.3
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.42.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by maheshmnj** [comment]
cc: @yjbanov

**Comment by snowb1shop** [comment]
Same issue

**Comment by OleFriedrich** [comment]
> Any update on this one? Or an idea for a workaround? I have a Flutter Web app in production and really needs this issue fixed

Same here. For the time being, we will force building with the html renderer for our web deployment, because here the problem does not occur. However, enforcing the use of the html renderer probably isn't such a good idea in the long run.

You can force the html renderer with `flutter build web --web-renderer html`

**Comment by yjbanov** [comment]
A workaround is to disable browser image codecs by passing `--dart-define=BROWSER_IMAGE_DECODING_ENABLED=false` to `flutter run` and `flutter build` (only works in profile and release modes). We'll need to investigate why this is happening.

**Comment by eyebrowsoffire** [comment]
After some investigation and discussion with Chromium folks, this is actually a bug in Chromium's implementation of ImageDecoder. It is being tracked here, and there is a fix out for PR: https://bugs.chromium.org/p/chromium/issues/detail?id=1342672

**Comment by eyebrowsoffire** [comment]
The fix has landed, and I confirmed it is fixed in Chrome 105.0.5168.0!

**Comment by dalecurtis** [comment]
The fix will also go out in the next M104 beta release and hit stable around ~Aug 2 per https://chromiumdash.appspot.com/schedule
